### ISSUE TITLE
Learning Model-Agnostic Meta-Learning for Fast Adaptation of Deep Networks

### ISSUE THREAD

**Issue by sweaterr** [sweaterr/readinglist#347]

**Comment by sweaterr** [comment]
## 3.1. Supervised Regression and Classification

Few-shot learning is well-studied in the domain of supervised tasks, where the goal is to learn a new function from only a few input/output pairs for that task, using prior data from similar tasks for meta-learning.
메타 - 학습을위한 유사한 태스크의 이전 데이터를 사용하여 태스크에 대한 몇 가지 입력 / 출력 쌍에서 새로운 기능을 학습하는 것이 목표 인 감독 된 태스크 영역에서 잘 학습되지는 않습니다.

For example, the goal might be to classify images of a Segway after seeing only one or a few examples of a Segway, with a model that has previously seen many other types of objects.
예를 들어, 이전에 많은 다른 유형의 객체를 본 모델과 함께 Segway의 하나 또는 몇 가지 예를 본 후에 Segway의 이미지를 분류하는 것이 목표 일 수 있습니다.

Likewise, in few-shot regression, the goal is to predict the outputs of a continuous-valued function from only a few datapoints sampled from that function, after training on many functions with similar statistical properties.
마찬가지로, 몇 번의 회귀 분석에서, 목표는 유사한 통계적 특성을 가진 많은 함수를 학습 한 후에 그 함수로부터 샘플링 된 소수의 데이터 포인트에서 연속 값 함수의 출력을 예측하는 것입니다.

**Comment by sweaterr** [comment]
Similarly, for discrete classification tasks with a crossentropy loss, the loss takes the form:.
유사하게, 교차 감쇠 손실을 갖는 이산 분류 작업의 경우 손실은 다음 형식을 취합니다.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/38845370-20a5f982-4232-11e8-9639-2609189f63bb.png)

**Comment by sweaterr** [comment]
According to the conventional terminology, K-shot classification tasks use K input/output pairs from each class, for a total of NK data points for N-way classification.
기존의 용어에 따르면, K- 샷 분류 작업은 N- 웨이 분류를위한 총 NK 데이터 포인트에 대해 각 클래스의 K 입력 / 출력 쌍을 사용합니다.

Given a distribution over tasks p(Ti), these loss functions can be directly inserted into the equations in Section 2.2 to perform meta-learning, as detailed in Algorithm 2.
태스크 p (Ti)에 대한 분포가 주어지면, 알고리즘 2에서 설명한 것처럼 이러한 손실 함수를 섹션 2.2의 방정식에 직접 삽입하여 메타 학습을 수행 할 수 있습니다.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/38845446-6ab6a422-4232-11e8-939f-27fe065f8388.png)

**Comment by sweaterr** [comment]
## 5. Experimental Evaluation

**Comment by sweaterr** [comment]
## 5.2. Classification

To evaluate MAML in comparison to prior meta-learning and few-shot learning algorithms, we applied our method to few-shot image recognition on the Omniglot (Lake et al,, 2011) and MiniImagenet datasets.
이전의 메타 학습 및 소수 학습 알고리즘과 비교하여 MAML을 평가하기 위해 Omniglot (Lake et al, 2011) 및 MiniImagenet 데이터 세트의 소수 이미지 인식에이 방법을 적용했습니다.

The Omniglot dataset consists of 20 instances of 1623 characters from 50 different alphabets.
Omniglot 데이터 세트는 50 개의 다른 알파벳으로 구성된 1623 개의 문자로 구성된 20 개의 인스턴스로 구성됩니다.

Each instance was drawn by a different person.
각 인스턴스는 다른 사람이 그렸습니다.

The MiniImagenet dataset was proposed by Ravi & Larochelle (2017), and involves 64 training classes, 12 validation classes, and 24 test classes.
MiniImagenet 데이터 세트는 Ravi & Larochelle (2017)에 의해 제안되었으며 64 개의 교육 클래스, 12 개의 검증 클래스 및 24 개의 테스트 클래스가 포함됩니다.

The Omniglot and MiniImagenet image recognition tasks are the most common recently used few-shot learning benchmarks (Vinyals et al,, 2016; Santoro et al,, 2016; Ravi & Larochelle, 2017).
Omniglot 및 MiniImagenet 이미지 인식 작업은 최근 가장 자주 사용되는 몇 가지 학습 학습 벤치 마크입니다 (Vinyals et al, 2016; Santoro et al, 2016; Ravi & Larochelle, 2017).

We follow the experimental protocol proposed by Vinyals et al, (2016), which involves fast learning of N-way classification with 1 or 5 shots.
우리는 Vinyals et al (2016)에 의해 제안 된 실험 프로토콜을 따르고, 1 또는 5 샷으로 N-way 분류법을 빨리 학습해야한다.

The problem of N-way classification is set up as follows: select N unseen classes, provide the model with K different instances of each of the N classes, and evaluate the model’s ability to classify new instances within the N classes.
N 방향 분류의 문제는 다음과 같이 설정됩니다. N 개의 보이지 않는 클래스를 선택하고, N 개의 클래스 각각에 대해 K 개의 다른 인스턴스를 모델에 제공하고, N 개의 클래스에서 새 인스턴스를 분류하는 모델의 기능을 평가합니다.

For Omniglot, we randomly select 1200 characters for training, irrespective of alphabet, and use the remaining for testing.
Omniglot의 경우 알파벳에 관계없이 무작위로 1200자를 선택하고 테스트를 위해 남은 문자를 사용합니다.

The Omniglot dataset is augmented with rotations by multiples of 90 degrees, as proposed by Santoro et al, (2016).
Omniglot 데이터 세트는 Santoro 외 (2016)에 의해 제안 된대로 90도 배수의 회전으로 보강됩니다.

**Comment by sweaterr** [comment]
Our model follows the same architecture as the embedding function used by Vinyals et al, (2016), which has 4 modules with a 3 × 3 convolutions and 64 filters, followed by batch normalization (Ioffe & Szegedy, 2015), a ReLU nonlinearity, and 2 × 2 max-pooling.
우리 모델은 Vinyals et al (2016)에서 사용 된 임베딩 함수와 동일한 아키텍처를 따르며, 3x3 컨볼 루션 및 64 필터가있는 4 개의 모듈이 있으며 배치 표준화 (Ioffe & Szegedy, 2015), ReLU 비선형 성, 및 2 × 2 최대 풀링.

The Omniglot images are downsampled to 28 × 28, so the dimensionality of the last hidden layer is 64.
Omniglot 이미지는 28 × 28로 다운 샘플링되므로 마지막 숨겨진 레이어의 차원은 64입니다.

As in the baseline classifier used by Vinyals et al, (2016), the last layer is fed into a softmax  For Omniglot, we used strided convolutions instead of max-pooling.
Vinyals et al (2016)에 의해 사용 된 baseline 분류기에서와 같이, 마지막 층은 softmax로 공급된다. Omniglot을 위해 우리는 max-pooling 대신 strided convolutions을 사용했다.

For MiniImagenet, we used 32 filters per layer to reduce overfitting, as done by (Ravi & Larochelle, 2017).
MiniImagenet의 경우 계층 당 32 개의 필터를 사용하여 overfitting을 줄였습니다 (Ravi & Larochelle, 2017).

In order to also provide a fair comparison against memory-augmented neural networks (Santoro et al,, 2016) and to test the flexibility of MAML, we also provide results for a non-convolutional network.
메모리 증가 신경망 (Santoro et al ,, 2016)과 공정한 비교를 제공하고 MAML의 유연성을 테스트하기 위해 비 갈비 네트워크에 대한 결과도 제공합니다.

For this, we use a network with 4 hidden layers with sizes 256, 128, 64, 64, each including batch normalization and ReLU nonlinearities, followed by a linear layer and softmax  For all models, the loss function is the cross-entropy error between the predicted and true class.
이를 위해 크기 256, 128, 64, 64의 숨겨진 레이어 4 개가있는 네트워크를 사용합니다. 각 레이어는 정규화 및 ReLU 비선형 성을 포함하고 선형 레이어와 softmax가 뒤에옵니다. 모든 모델에서 손실 함수는 다음과 같은 교차 엔트로피 오류입니다. 예측되고 참된 계급.

Additional hyperparameter details are included in Appendix A.1.
추가적인 hyperparameter 세부 사항은 부록 A.1에 포함되어있다.

**Comment by sweaterr** [comment]
We present the results in Table 1.
표 1에 결과를 나타냅니다.

The convolutional model learned by MAML compares well to the state-of-the-art results on this task, narrowly outperforming the prior methods.
MAML이 배운 길쌈 모델은이 작업에 대한 최첨단 결과와 잘 비교되며, 이전 방법을 능가하는 성능을 크게 능가합니다.

Some of these existing methods, such as matching networks, Siamese networks, and memory models are designed with few-shot classification in mind, and are not readily applicable to domains such as reinforcement learning.
매칭 네트워크, 샴 네트워크 및 메모리 모델과 같은 기존 방법 중 몇 가지를 염두에두고 몇 가지 분류를 염두에두고 설계되었으며 강화 학습과 같은 도메인에는 쉽게 적용 할 수 없습니다.

Additionally, the model learned with MAML uses fewer overall parameters compared to matching networks and the meta-learner LSTM, since the algorithm does not introduce any additional parameters beyond the weights of the classifier itself.
또한 MAML로 학습 한 모델은 분류기 자체의 가중치 이상으로 추가 매개 변수를 도입하지 않으므로 일치하는 네트워크 및 메타 학습자 LSTM에 비해 전체 매개 변수가 적습니다.

Compared to these prior methods, memory-augmented neural networks (Santoro et al,, 2016) specifically, and recurrent meta-learning models in general, represent a more broadly applicable class of methods that, like MAML, can be used for other tasks such as reinforcement learning (Duan et al,, 2016b; Wang et al,, 2016).
이러한 이전의 방법들과 비교할 때, 기억 증강 신경망 (Santoro et al, 2016) 및 반복적 인 메타 학습 모델은 일반적으로 MAML과 같은 다른 작업에 사용될 수있는 좀 더 광범위하게 적용 가능한 방법의 분류를 나타낸다 (Duan et al, 2016b; Wang et al, 2016).

However, as shown in the comparison, MAML significantly outperforms memory-augmented networks and the meta-learner LSTM on 5-way Omniglot and MiniImagenet classification, both in the 1-shot and 5-shot case.
그러나 비교에서 볼 수 있듯이 MAML은 1 회 및 5 회 모두 5 비트 Omniglot 및 MiniImagenet 분류에서 메모리 확장 네트워크와 메타 학습자 LSTM보다 월등히 뛰어납니다.

**Comment by sweaterr** [comment]
A significant computational expense in MAML comes from the use of second derivatives when backpropagating the meta-gradient through the gradient operator in the meta-objective (see Equation (1)).
MAML의 중요한 계산 비용은 메타 목표에서 그라디언트 연산자를 통해 메타 그래디언트를 백 프로퍼 그 레이션 할 때 두 번째 파생물을 사용하는 것입니다 (방정식 (1) 참조).

On MiniImagenet, we show a comparison to a first-order approximation of MAML, where these second derivatives are omitted.
MiniImagenet에서 MAML의 1 차 근사법과의 비교를 보여줍니다.이 2 차 미분은 생략됩니다.

Note that the resulting method still computes the meta-gradient at the post-update parameter values θ 0 i , which provides for effective meta-learning.
결과 메쏘드는 효과적인 메타 - 학습을 제공하는 업데이트 후 파라미터 값 θ0i에서 메타 - 그래디언트를 계산한다.

Surprisingly however, the performance of this method is nearly the same as that obtained with full second derivatives, suggesting that most of the improvement in MAML comes from the gradients of the objective at the post-update parameter values, rather than the second order updates from differentiating through the gradient update.
그러나 놀랍게도이 방법의 성능은 전체 2 차 미분 계수에서 얻은 결과와 거의 동일하므로 MAML의 개선 된 부분은 2 차 업데이트가 아닌 업데이트 후 매개 변수 값의 목표 기울기에서 비롯된 것입니다. 그래디언트 업데이트를 통해 차별화됩니다.

Past work has observed that ReLU neural networks are locally almost linear (Goodfellow et al,, 2015), which suggests that second derivatives may be close to zero in most cases, partially explaining the good perfor mance of the first-order approximation.
과거의 연구는 ReLU 신경 회로망이 국부적으로 거의 직선형임을 보여 주었는데 (Goodfellow et al ,, 2015), 2 계 도함수가 대부분 0에 가까울 수 있다고 제안하여 1 차 근사법의 좋은 성능을 부분적으로 설명했다.

This approximation removes the need for computing Hessian-vector products in an additional backward pass, which we foun.
이 근사법은 추가 역방향 패스에서 헤 시안 벡터 곱을 계산할 필요가 없도록 만든 것입니다.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/38846308-ba472bd4-4236-11e8-81f1-c7a41f18ff26.png)
